sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CSharpInvocationTracker.cs

```diff
@@ -47,7 +46,19 @@ public CSharpInvocationTracker(IAnalyzerConfiguration analysisConfiguration, Dia
         public override InvocationCondition MatchSimpleNames(params MethodSignature[] methods)
         {
             return (context) => MethodSignatureHelper.IsMatch(context.Identifier as SimpleNameSyntax,
-                    context.Model, context.InvokedMethodSymbol, methods);
+                    context.Model, context.InvokedMethodSymbol, true, methods);
+        }
+
+        public override bool FirstParameterIsStringAndIsNotConstant(InvocationContext context)
```

Comment:
New check required by the "executing XPath" rule.
I've added an _IsConstant_ extension method to the _xxxSyntaxHelper_ classes.